src/swarm-tui/Swarm/TUI/Controller.hs

```diff
@@ -327,50 +328,53 @@ handleMainEvent forceRedraw ev = do
     VtyEvent vev
       | isJust (s ^. playState . uiGameplay . uiDialogs . uiModal) -> handleModalEvent vev
     MouseDown (TerrainListItem pos) V.BLeft _ _ ->
-      playState . uiGameplay . uiWorldEditor . terrainList %= BL.listMoveTo pos
+      Brick.zoom playState $ uiGameplay . uiWorldEditor . terrainList %= BL.listMoveTo pos
     MouseDown (EntityPaintListItem pos) V.BLeft _ _ ->
-      playState . uiGameplay . uiWorldEditor . entityPaintList %= BL.listMoveTo pos
-    MouseDown WorldPositionIndicator _ _ _ -> playState . uiGameplay . uiWorldCursor .= Nothing
+      Brick.zoom playState $ uiGameplay . uiWorldEditor . entityPaintList %= BL.listMoveTo pos
+    MouseDown WorldPositionIndicator _ _ _ ->
+      Brick.zoom playState $ uiGameplay . uiWorldCursor .= Nothing
     MouseDown (FocusablePanel WorldPanel) V.BMiddle _ mouseLoc ->
       -- Eye Dropper tool
-      EC.handleMiddleClick mouseLoc
+      Brick.zoom playState $ EC.handleMiddleClick mouseLoc
```

Comment:
Are these calls to `Brick.zoom` necessary?

**Edited to add**: Oh, wait, I see, the type of e.g. `EC.handleMiddleClick` changed from `... -> EventM Name AppState ()` to `... -> EventM Name PlayState ()`.  Makes sense.